### PR TITLE
Mtw 65 Template homescreen pod component, fetch data with query string & Reuse code in backend

### PR DESCRIPTION
**Pull request by irina-wang**
**HomeScreenPod.js**

- contain template for Home Screen pod component to account for future changes to avoid code repeat.
- fetch data using the **GET** method sending the query _pod=${pod}_ to the backend

**app.py**

- Extract pod from the request arguments from frontend
- reuse code to get data in sf for each pod
- send back only progress and total outcomes for the current pod

> //  previous changes(reverted)
> - Added to the outcomes endpoint in app.py to calculate the total number of outcomes and then used that info in the frontend of HomeScreen.js for the progress bars; should still work if new outcomes are added
> - Use desc and "Outcomes__c" to filter related field
> - transform json data to python dictionary(easier to process data without hardcoding) 
> - process data to get the sum for all outcomes in the Trainee pod

**HomeScreen.js**

- create an array that contains all the pod names
- render the three pods on the home screen by passing data to HomeScreenPod 

> //  previous changes(reverted)
> - Change to using real user data instead of fake user data by passing in the user parameter in app.py and by using token in HomeScreen.js (also changed from axios to fetch).

---------------------------------------------------------------------------------------

Steps to test PR:
Login in on app using Jumbo1's account to test having real user data:

Username: titapa.punpun@gmail.com      
password: JumboOne!!

Then 
- the Home Screen should load all the progress bars with the correct number of total outcomes
- The first pod(Trainee) should be highlighted
- Navigation to the Trainee, Associate, and Partner all work fine

> Note:
> Css highlight logic: when _complete_ is smaller than _total_ (ongoing) and is _not zero_(have started)
> -  it won't work for some of the current mock data, but it will work in real user scenario -- when user are not allowed to start next pod (i.e have complete outcome with the new pod) unless they finish the previous one.

---------------------------------------------------------------------------------------

Standard Checklist:
- [X] Does the code work? Does it perform its intended function, the logic is correct etc.
- [X] Have you merged the most recent version of staging into your branch?
- [X] Are all merge conflicts resolved? 
- [X] Can any logging or debugging code be removed? (Any debugging statements should be removed)
- [X] Is the `diff` the **minimum amount** needed to achieve this ticket's goal?

### CODE REVIEW

File: backend/app.py [hunk 1]
```diff
@@ -91,40 +90,6 @@ def sample(user):
...
-        })
-        response.status_code = 401
-        return response
-    
-    # update the contact
-    sf.Contact.update(response["records"][0]["Id"], {"Has_Youth_App_Account__c": True})
-
-    return jsonify({"result": "success"})
-
-
```
Comment (jchin01) [inline backend/app.py]: Why are you deleting this endpoint? is it not longer used for authentication? check with yichen to double check please

File: backend/app.py [hunk 2]
```diff
@@ -146,52 +111,59 @@ def verify():
...
 
     # salesforce query based on the email, firstname & lastname
-    result = sf.query(
-        format_soql(
-            "SELECT Id, Email FROM Contact WHERE (MTW_Role__c = 'MTW Young Adult' AND email = {email_value} AND name={full_name})", 
-            email_value=email, 
-            full_name=name))
-
-    print(result["totalSize"] == 1)
+    result = sf.query(format_soql("SELECT Id, Email FROM Contact WHERE (email = {email_value} AND name={full_name})", email_value=email, full_name=name))
```
Comment (jchin01) [inline backend/app.py]: Also why are there changes to the verify sign up endpoint?

**Review by jchin01 — CHANGES_REQUESTED** [review]

**Comment by titapa-punpun** [comment]
@irina-wang Please change the "x of x outcomes achieved" text to white to match Abe's design, thanks!

**Comment by irina-wang** [comment]
I implemented greying out, changed the text color accordingly, and tested for complete, ongoing, and grey states. The only thing that differs from Abe's design is the bar fill color for ongoing pods. I made all with white background and green fill because the code becomes messy when we want to change [animated styling](https://docs.swmansion.com/react-native-reanimated/docs/next/api/useAnimatedProps/#adapter-function). I will talk to Abe and see what he prefers. 

Question: what should I do for the pods when there's no data was fetched from the backend(for Jumbo 4)?

**Comment by titapa-punpun** [comment]
Reopen PR once merge conflicts have been resolved and when finished pod color is changed to be a lighter shade of grey.

**Pull request closed without merging**